src/AutoloadGenerator.php

```diff
@@ -2,30 +2,268 @@
 
 namespace ComposerWordPressAutoloader;
 
+use Composer\Composer;
 use Composer\Autoload\AutoloadGenerator as ComposerAutoloadGenerator;
+use Composer\IO\IOInterface;
 use Composer\Package\PackageInterface;
+use Composer\Package\RootPackageInterface;
+use Composer\Util\Filesystem;
 
 /**
  * Composer Autoload Generator
  */
 class AutoloadGenerator extends ComposerAutoloadGenerator
 {
-  /**
-   * Compiles an ordered list of namespace => path mappings
-   *
-   * @param array $packageMap
-   * @param PackageInterface $rootPackage
-   * @param boolean $filteredDevPackages
-   * @return array
-   */
+    protected Composer $composer;
+    protected bool $devMode = true;
+
+    /**
+     * Constructor.
+     *
+     * @param Composer $composer
+     * @param IOInterface $io
+     */
+    public function __construct(Composer $composer, IOInterface $io)
+    {
+        parent::__construct($composer->getEventDispatcher(), $io);
+
+        $this->composer = $composer;
+    }
+
+    /**
+     * @param bool $devMode
+     * @return void
+     */
+    public function setDevMode($devMode = true)
+    {
+        parent::setDevMode($devMode);
+
+        $this->devMode = (bool) $devMode;
+    }
+
+    /**
+     * Generate the autoload file.
+     *
+     * @param boolean $beingInjected Flag if the autoload file is being injected.
+     * @param boolean $isDevMode Flag if dev dependencies are being included.
+     * @return string
+     */
+    public function generate(bool $beingInjected, bool $isDevMode = true): string
+    {
+        $filesystem = new Filesystem();
+        $autoloadFileContents = '';
+
+        $basePath = $filesystem->normalizePath(realpath(realpath(getcwd())));
+        $vendorPath = $filesystem->normalizePath(realpath(realpath($this->composer->getConfig()->get('vendor-dir'))));
```

Comment:
I assume calling `realpath` twice on each is intentional?

Reply:
It is. It's copy+paste from the composer code that did the same